core/src/com/unciv/logic/map/tile/Tile.kt

```diff
@@ -168,6 +186,10 @@ class Tile : IsPartOfGameInfoSerialization {
     private var isAdjacentToRiver = false
     @Transient
     private var isAdjacentToRiverKnown = false
+
+    val improvementInProgress get() = improvementQueue.firstOrNull()?.improvement
```

Comment:
We have: improvementInProgress, turnsToImprovement, hasImprovementInProgress and getTileImprovementInProgress() in this class. Are the last two nessessary? If so can you move them to be together?

Reply:
I meant more like deleting getTileImprovementInProgress() and moving hasImprovementInProgress() under turnsToImprovement.